src/math/mat4.js

```diff
@@ -1225,6 +1225,9 @@ class Mat4 {
         sy = scale.y;
         sz = scale.z;
 
+        if (sx === 0 || sy === 0)
```

Comment:
Don't you also have to check whether `z` is `0`?

Reply:
Yes, I do, will update.